### DISCUSSION TITLE
Yank or Copy content of doHover

### DISCUSSION THREAD

**Discussion by dorukozerr** [neoclide/coc.nvim#5157] [Q&A] [answered]
Hello everyone, I'm new to Vim and its plugins. 

I configured coc and it's working great. One thing I couldn't find how to do is I can't copy or yank the content of coc doHover action. For example, I'm working on a TS project, and let's say I want to copy the type definition of a variable. But I can't do it. I don't want to set mouse=a and do it with mouse. Can anyone help me pretty please :) I searched about this but couldn't find anything. I found 1 GitHub post about this but it doesn't work. 

Thank you so much for your attention and participation.

Example Img

<img width="387" alt="image" src="https://github.com/user-attachments/assets/2091c78d-21e3-4a0a-bb08-61fdc63f1337">

**Comment by fannheyward** [comment]
You can use `Ctrl-ww` to move your cursor to the floating window, to copy the content.

**Reply by dorukozerr** [reply]
Thanks for the reply but it doesn't work, I can't move the cursor to the floating window with `ctrl-ww`.

**Reply by fannheyward** [reply]
I forgot that `C-ww` only works on nvim. Vim doesn't have a way to jump into the floating window.

**Reply by dorukozerr** [reply]
Maybe I should switch to nvim from vim. I wanted to learn the default basics before jumping into it. Thanks for replys.

**Reply by dorukozerr** [reply, marked as answer]
`tsc filename.ts --declaration --emitDeclarationOnly`

This generates what I need, and I am posting this for a solution. [Stack overflow solution](https://stackoverflow.com/a/68990816)

**Comment by chemzqm** [comment]
You can create a function to copy popup content, like:
```vim
  function! CopyPopupBufferToClipboard()
    let buffers = getbufinfo({'bufloaded': 1, 'buftype': 'nofile'})
    for buf in buffers
      if has_key(buf, 'popups') && len(buf.popups) > 0
        let lines = getbufline(buf['bufnr'], 1, '$')
        call setreg('*', lines)
        echo "已将 popup 窗口的内容复制到系统剪贴板。"
        return
      endif
    endfor
    echo "未找到任何 popup 窗口。"
  endfunction
```
Chekcout `:h getbufinfo()` for support of `popups`

**Reply by dorukozerr** [reply]
Wow that's nice, Thanks a lot.
